pulp_rpm/app/shared_utils.py

```diff
@@ -24,12 +25,18 @@ def _prepare_package(artifact, filename):
         filename: name of file uploaded by user
     """
     artifact_file = storage.open(artifact.file.name)
+    cr_pkginfo = createrepo_c.package_from_rpm(
```

Comment:
If I remember correctly, the problem was that createrepo wouldn't parse the file at all unless it had a `.rpm` extension.  This is why we renamed the file prior to parsing it.  So I'm unclear on whether this would actually work if you cannot parse the file before renaming it.

If you could, this would render the rename operation pointless as it was the only reason the rename was done.

What might work fine, is if it was just named as `foo.rpm` or something along those lines.

Reply:
ensure a file has an rpm extension.